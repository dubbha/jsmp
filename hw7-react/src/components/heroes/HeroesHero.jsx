import React from 'react';
import PropTypes from 'prop-types';

const HeroesHero = ({ hero, setSelectedHero, className }) => (
  <li onClick={setSelectedHero} className={className}>
    <span className="badge">{hero.id}</span> {hero.name}
  </li>
);

HeroesHero.propTypes = {
  hero: PropTypes.object.isRequired,
  setSelectedHero: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
};

export default HeroesHero;
