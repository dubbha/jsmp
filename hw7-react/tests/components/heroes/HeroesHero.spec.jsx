import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HeroesHero from 'components/heroes/HeroesHero';

describe('HeroesHero', () => {
  it('should render hero element in the heroes list', () => {
    const props = {
      hero: { id: 777, name: 'Zero' },
      setSelectedHero: jest.fn(),
      className: 'block__element_modifier',
    };
    const wrapper = shallow(<HeroesHero {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
