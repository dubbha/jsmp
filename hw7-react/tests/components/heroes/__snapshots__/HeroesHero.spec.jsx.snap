// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesHero should render hero element in the heroes list 1`] = `
<li
  className="block__element_modifier"
  onClick={[Function]}
>
  <span
    className="badge"
  >
    777
  </span>
   
  Zero
</li>
`;
